library/pk.c

```diff
@@ -1404,10 +1396,12 @@ int mbedtls_pk_setup( mbedtls_pk_context *ctx, mbedtls_pk_handle_t info )
         return( MBEDTLS_ERR_PK_BAD_INPUT_DATA );
 
     ctx->pk_info = info;
-#endif
 
     if( ( ctx->pk_ctx = pk_info_ctx_alloc_func( info ) ) == NULL )
         return( MBEDTLS_ERR_PK_ALLOC_FAILED );
+#else
```

Comment:
Major: I think we should either document that it is assumed that zeroization is sufficient for initialization and freeing in case `MBEDTLS_PK_SINGLE_TYPE` is used, or introduce separate `init/free` functions in this case.